server/src/main/java/org/elasticsearch/snapshots/RepositoryIntegrityHealthIndicatorService.java

```diff
@@ -0,0 +1,102 @@
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.snapshots;
+
+import org.elasticsearch.cluster.metadata.RepositoriesMetadata;
+import org.elasticsearch.cluster.metadata.RepositoryMetadata;
+import org.elasticsearch.cluster.service.ClusterService;
+import org.elasticsearch.health.HealthIndicatorDetails;
+import org.elasticsearch.health.HealthIndicatorResult;
+import org.elasticsearch.health.HealthIndicatorService;
+import org.elasticsearch.health.SimpleHealthIndicatorDetails;
+import org.elasticsearch.repositories.RepositoryData;
+
+import java.util.List;
+import java.util.Map;
+
+import static org.elasticsearch.common.Strings.collectionToDelimitedStringWithLimit;
+import static org.elasticsearch.health.HealthStatus.GREEN;
+import static org.elasticsearch.health.HealthStatus.RED;
+import static org.elasticsearch.health.ServerHealthComponents.SNAPSHOT;
+
+/**
+ * This indicator reports health for snapshot repositories.
+ *
+ * Indicator will report RED status when any of snapshot repositories is marked as corrupted.
+ * Data might not be backed up in such cases.
+ *
+ * Corrupted repository most likely need to be manually cleaned and a new snapshot needs to be created from scratch.
+ */
+public class RepositoryIntegrityHealthIndicatorService implements HealthIndicatorService {
+
+    public static final String NAME = "repository integrity";
+
+    private final ClusterService clusterService;
+
+    public RepositoryIntegrityHealthIndicatorService(ClusterService clusterService) {
+        this.clusterService = clusterService;
+    }
+
+    @Override
+    public String name() {
+        return NAME;
+    }
+
+    @Override
+    public String component() {
+        return SNAPSHOT;
+    }
+
+    @Override
+    public HealthIndicatorResult calculate() {
+        var snapshotMetadata = clusterService.state().metadata().custom(RepositoriesMetadata.TYPE, RepositoriesMetadata.EMPTY);
+
+        if (snapshotMetadata.repositories().isEmpty()) {
+            return createIndicator(GREEN, "No repositories configured", HealthIndicatorDetails.EMPTY);
+        }
+
+        var corrupted = snapshotMetadata.repositories()
+            .stream()
+            .filter(repository -> repository.generation() == RepositoryData.CORRUPTED_REPO_GEN)
+            .map(RepositoryMetadata::name)
+            .toList();
+
+        var totalRepositories = snapshotMetadata.repositories().size();
+        var corruptedRepositories = corrupted.size();
+
+        if (corrupted.isEmpty()) {
+            return createIndicator(
+                GREEN,
+                "No corrupted repositories",
+                new SimpleHealthIndicatorDetails(Map.of("total-repositories", totalRepositories))
```

Comment:
I think we are going with underscored opposed to hyphenated. At least that is what I did in the first PR based on the other cluster health API.

Reply:
This comment applies to pretty much every place where words are hyphenated.

Reply:
Good point, looks like we are consistently using `_` across all APIs. Will fix that